### PR TITLE
Add front end web project

### PR DESCRIPTION
**Pull request by daniefer**
Contains a bunch of default views from the Visual Studio MVC project template, including security. Flushed out a rough layout for the notes view. Several other stories/tasks will fall out of this commit. The web api is not fully functional at this point. Have some issues getting and creating new notes. Will split #2 into sub tasks, since it was probably too large to do in one sitting anyways.

### CODE REVIEW

**Comment by daniefer** [comment]
@legendofcolt and @jppotess - Here are the commits for #1 and #2. I'm trying my best at ITCSS/BEMIT (i'm not to sure which style I am following). A few questions I have about my approach are:
- How do I build off the bootstrap for my theme, or do I have to statrt over if I am using BEM?
- Do either of you have any good resources over how to name theme variables? For instance: I have `$text-primary-color` and `$logo-primary-color` which are used to set the colors throughout the site. Any ideas on how to do that without naming the variables `$dark-blue-used-for-header-text-on-landing-page`, etc. I don't understand what primary/secondary mean from a design standpoint, I've just heard its a best practice.

**Comment by cy-by** [comment]
Wow, that's one big commit you've got here!

Bootstrap doesn't use the BEM methodology. But I also wouldn't worry about. Just be aware that it's different. BEM is suppose to help with styles be as consistent to avoid several CSS failings as a language. ITCSS is how to structure your CSS files for large teams/web apps. Since it's only you, I'd say you can decided on how much you want to follow it or not.

`$text-primary-color` is a good naming scheme in the sense that if I come across it in a SCSS file and have never seen it before, I know its purpose. To color the text with the primary color. Naming things are very difficult, for example that color could also be used for a header background. Now the variable name isn't very descriptive. One reason why you hear that `primary` and `secondary` are good words to use is because if you say `$myFavoriteBlue` as your primary color of the entire app. But then rebrand to be yellow, then your variable name would be very confusing.

Basically, just write what makes sense to you, try not to over think it, and when your working with a team, be open to change.

I enjoy your weekend hustle 👍  👍  👍
